shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-metrics/shardingsphere-agent-metrics-api/src/main/java/org/apache/shardingsphere/agent/metrics/api/definition/MetricsPluginDefinitionService.java

```diff
@@ -17,58 +17,56 @@
-    
-    private static final String TRANSACTION_ENHANCE_CLASS = "org.apache.shardingsphere.proxy.backend.communication.jdbc.transaction.BackendTransactionManager";
-    
-    private static final String TRANSACTION_ADVICE_CLASS = "org.apache.shardingsphere.agent.metrics.api.advice.TransactionAdvice";
-    
-    private static final String DATASOURCE_ENHANCE_CLASS = "org.apache.shardingsphere.infra.config.datasource.JDBCParameterDecoratorHelper";
-    
-    private static final String DATASOURCE_ADVICE_CLASS = "org.apache.shardingsphere.agent.metrics.api.advice.DataSourceAdvice";
-    
     @Override
     public void defineInterceptors() {
-        defineInterceptor(COMMAND_EXECUTOR_TASK_ENHANCE_CLASS)
-                .aroundInstanceMethod(ElementMatchers.named(MethodNameConstant.COMMAND_EXECUTOR_RUN))
-                .implement(COMMAND_EXECUTOR_TASK_ADVICE_CLASS)
-                .build();
-        defineInterceptor(CHANNEL_HANDLER_ENHANCE_CLASS)
-                .aroundInstanceMethod(ElementMatchers.named(MethodNameConstant.CHANNEL_ACTIVE).or(ElementMatchers.named(MethodNameConstant.CHANNEL_INACTIVE))
-                        .or(ElementMatchers.named(MethodNameConstant.CHANNEL_READ)))
-                .implement(CHANNEL_HANDLER_ADVICE_CLASS)
-                .build();
-        defineInterceptor(SQL_ROUTER_ENGINE_ENHANCE_CLASS)
-                .aroundInstanceMethod(ElementMatchers.named(MethodNameConstant.SQL_ROUTER))
-                .implement(SQL_ROUTER_ENGINE_ADVICE_CLASS)
-                .build();
-        defineInterceptor(TRANSACTION_ENHANCE_CLASS)
-                .aroundInstanceMethod(ElementMatchers.named(MethodNameConstant.COMMIT).or(ElementMatchers.named(MethodNameConstant.ROLL_BACK)))
-                .implement(TRANSACTION_ADVICE_CLASS)
-                .build();
-        defineInterceptor(DATASOURCE_ENHANCE_CLASS)
-                .aroundClassStaticMethod(ElementMatchers.named(MethodNameConstant.DECORATE))
-                .implement(DATASOURCE_ADVICE_CLASS)
-                .build();
+        Yaml yaml = new Yaml();
+        InputStream in = this.getClass().getResourceAsStream("/interceptors.yaml");
+        Interceptors interceptors = yaml.loadAs(in, Interceptors.class);
+        for (Interceptor interceptor : interceptors.getInterceptors()) { 
+            String[] instancePoints = interceptor
+                    .getPoints()
+                    .stream()
+                    .filter(i -> i.getType().equals("instance"))
+                    .map(TargetPoint::getName)
+                    .toArray(String[]::new);
+            String[] staticPoints = interceptor
+                    .getPoints()
+                    .stream()
+                    .filter(i -> i.getType().equals("static"))
+                    .map(TargetPoint::getName)
+                    .toArray(String[]::new);
+    
```

Comment:
It's suggested to remove this blank line. If you want to split these codings, extracting a function is a preferred way.

Reply:
ok